runtime/interpreter/interpreter.go

```diff
@@ -1107,6 +1161,23 @@ func (interpreter *Interpreter) declareNonEnumCompositeValue(
 				locationRange := invocation.LocationRange
 				self := *invocation.Self
 
+				if len(compositeType.ConstructorParameters) < 1 {
+					return nil
+				}
+
+				// event interfaces do not exist
+				compositeDecl := declaration.(*ast.CompositeDeclaration)
+				if compositeDecl.IsResourceDestructionDefaultEvent() {
+					// we implicitly pass the containing composite value as an argument to this invocation
+					containerComposite := invocation.Arguments[0].(*CompositeValue)
+					interpreter.activations.PushNewWithParent(inter.activations.CurrentOrNew())
+					interpreter.SharedState.callStack.Push(invocation)
```

Comment:
Maybe move this into `evaluateDefaultDestroyEvent` if possible, to keep the push and pop in the same place